deeppavlov/models/preprocessors/ner_preprocessor.py

```diff
@@ -0,0 +1,195 @@
+import errno
+import numpy as np
+import os
+from logging import getLogger
+from typing import List
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.models.estimator import Estimator
+
+log = getLogger(__name__)
+
+
+@register("ner_preprocessor")
+class ner_preprocessor():
+    def __init__(self, get_x_padded_for_elmo=False, get_x_cap_padded=False, **kwargs):
+        self.get_x_padded_for_elmo = get_x_padded_for_elmo
+        self.get_x_cap_padded = get_x_cap_padded
+
+        self.cap_vocab_size = 5
+
+    def encode_cap(self, s):
+        if s.upper() == s:
+            return 1
+        elif s.lower() == s:
+            return 2
+        elif (s[0].upper() == s[0]) and (s[1:].lower() == s[1:]):
+            return 3
+        else:
+            return 4
+
+    def __call__(self, batch: List[List[str]], **kwargs):
+        """
+        take as input batch of sents
+        Parameters
+        ----------
+        batch: list of list of tokens
+
+        Returns
+        -------
+        x_lower: batch in lowercase
+        sent_lengths: lengths of sents
+        x_padded_for_elmo (optional): batch padded with "", used as input for ELMo
+        x_cap_padded: batch of capitalization features
```

Comment:
There are components for lowercasing and getting capitalization features.
there is also a feature of padding with zeroes in `simple_vocab`

Reply:
This is used for ELMo and different from zero_padding in simple vocab. This paid sentences by addding empty word. I prefer including all of them in one function since they take as the same input (x_tokens). It looks clearer and simpler.